### PR TITLE
fix: package/ignition-math 

### PR DESCRIPTION
**Pull request by ggulgulia**
## Description
* The current recipe of ignition-math has bugs
   - it doesn't provide `Eigen` compoent
   - It doesn't set the ignition-math cmake VERSION info which is used in downstream dependencies
   - It uses the outdated ignition-math urls
       - Look at this post from gazebo on more updates: https://discourse.ros.org/t/a-new-era-for-gazebo-cross-post/25012

## Fix summary
* This MR fixes the issues described above
* Both the versions 6.7.0 and 6.10.0 are fixed in this MR

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ignition-math/all/conanfile.py [hunk 1]
```diff
@@ -104,46 +105,56 @@ def package(self):
...
+        self.cpp_info.names["cmake_paths"] = lib_name
+
+        self.cpp_info.components[lib_name].names["cmake_find_package"] = lib_name
+        self.cpp_info.components[lib_name].names["cmake_find_package_multi"] = lib_name
+        self.cpp_info.components[lib_name].names["cmake_paths"] = lib_name
+        self.cpp_info.components[lib_name].libs = [lib_name]
+        self.cpp_info.components[lib_name].includedirs.append(os.path.join("include", "ignition", "math"+version_major))
+        self.cpp_info.components[lib_name].requires = ["swig::swig", "eigen::eigen", "doxygen::doxygen"]
+        self.cpp_info.components[lib_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.components[lib_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```
Comment (madebr) [inline recipes/ignition-math/all/conanfile.py:142]: I'm not 100% sure this suggestion will fix the hook warnings.
```suggestion
        self.cpp_info.components[lib_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].builddirs["cmake_find_package"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].builddirs["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Comment (ggulgulia) [reply recipes/ignition-math/all/conanfile.py:148]: thanks for the suggestion. I applied this manually.  waiting for the CI build

Comment (ggulgulia) [reply recipes/ignition-math/all/conanfile.py:148]: @madebr your suggestions fixed the hooks warnings. Thanks :smile:

File: recipes/ignition-math/all/conanfile.py [hunk 2]
```diff
@@ -104,46 +105,63 @@ def package(self):
...
-        self.cpp_info.components["libignition-math-all"].requires.append("swig::swig")
-
-        self.env_info.LD_LIBRARY_PATH.extend([
-            os.path.join(self.package_folder, x) for x in self.cpp_info.libdirs
-        ])
-        self.env_info.PATH.extend([
-            os.path.join(self.package_folder, x) for x in self.cpp_info.bindirs
-        ])
+        lib_name = f"ignition-math{version_major}"
+        build_dirs = os.path.join("lib", "cmake")
```
Comment (madebr) [inline recipes/ignition-math/all/conanfile.py:131]: In my comment about hard coding the path, I meant to return `os.path.join("lib", "cmake")` in a method e.g. called `_module_dir_rel_path`. And let `_module_file_rel_path` use that method.

Comment (ggulgulia) [reply recipes/ignition-math/all/conanfile.py]: @madebr . Now does my change reflect the intent ?

Comment (madebr) [reply recipes/ignition-math/all/conanfile.py]: Excellent! Thanks!

**Review by uilianries — APPROVED** [review]

File: recipes/ignition-math/all/conanfile.py [hunk 3]
```diff
@@ -77,11 +78,11 @@ def build_requirements(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
```
Comment (uilianries) [inline recipes/ignition-math/all/conanfile.py:80]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
So you can remove the rename step

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**